### PR TITLE
Migrate Formstack Baton Lambda build process to Github Actions

### PR DESCRIPTION
**Pull request by akinsola-guardian**
## What does this change?
This migrates the deploy process from TeamCity to Github actions.

## How to test
Performing a push to check if RiffRaff detects the update, followed by deploying to the CODE environment.

## Accessibility

-   [ ] [Tested with screen reader](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#screen-reader)
-   [ ] [Navigable with keyboard](https://github.com/guardian/accessibility/blob/main/people-and-technology/02-physical.md#keyboard)
-   [ ] [Colour contrast passed](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#contrast)
-   [ ] [The change doesn't use only colour to convey meaning](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#use-of-colour)

### CODE REVIEW

File: .github/workflows/deploy-formstack.yml
Comment (kelvin-chappell) [inline .github/workflows/deploy-formstack.yml:16]: Perhaps add a note to say why these permissions are needed

File: .github/workflows/deploy-formstack.yml
Comment (kelvin-chappell) [inline .github/workflows/deploy-formstack.yml:32]: Doesn't `riffRaffUpload` in the previous line upload to Riffraff?  I'm asking as I've never used `actions-riff-raff` so I'm a bit confused

File: .github/workflows/deploy-formstack.yml
Comment (Mark-McCracken) [inline .github/workflows/deploy-formstack.yml]:
```suggestion
      - uses: actions/checkout@v3
```

**Review by shtukas — DISMISSED** [review]

File: .github/workflows/deploy-formstack.yml
Comment (Mark-McCracken) [inline .github/workflows/deploy-formstack.yml]:
```suggestion
      - uses: aws-actions/configure-aws-credentials@v2
```

File: .github/workflows/deploy-formstack.yml
Comment (Mark-McCracken) [reply .github/workflows/deploy-formstack.yml:32]: If it were me, I'd just leave off the `riffRaffArtifactResources`, and then get rid of the sbt plugin for riff raff.

**Review by Mark-McCracken — DISMISSED** [review]
Just a few tiny points to note, but these 2 version things don't actually have an impact if I remember correctly.
The riffRaffArtifactResources, I leave it up to you how you wanna deal with that.
But this would be fine to merge as it is, or if the only changes you make are around those 3 areas, feel free to merge with future commits without further approval.

File: .github/workflows/deploy-formstack.yml
Comment (kelvin-chappell) [reply .github/workflows/deploy-formstack.yml:32]: Ah I see.  So they both do the same thing but one's through SBT and the other's through GHA.

**Review by Mark-McCracken — APPROVED** [review]

File: .github/workflows/deploy-formstack.yml
Comment (Mark-McCracken) [reply .github/workflows/deploy-formstack.yml:32]: Yeah exactly. Little bit of a faff in github actions if you're not aware of whether it needs to produce a .jar or .deb file, and work out where it needs to land in the riffRaff folder structure, but hopefully once converted from TC to GHA, it'll be more clear to those who come to a project afresh

**Pull request merged**
